libarchive/archive_read_support_format_tar.c

```diff
@@ -1036,16 +1036,14 @@ header_Solaris_ACL(struct archive_read *a, struct tar *tar,
 	int64_t type;
 	char *acl, *p;
 
-	/*
-	 * read_body_to_string adds a NUL terminator, but we need a little
-	 * more to make sure that we don't overrun acl_text later.
-	 */
 	header = (const struct archive_entry_header_ustar *)h;
 	size = (size_t)tar_atol(header->size, sizeof(header->size));
 	archive_string_init(&acl_text);
 	err = read_body_to_string(a, tar, &acl_text, h, unconsumed);
-	if (err != ARCHIVE_OK)
+	if (err != ARCHIVE_OK) {
+		archive_string_free(&acl_text);
```

Comment:
Every other return in this function already cleans up the local `acl_text`.  But this spot was missed.